xarray/tests/test_dataset.py

```diff
@@ -1142,7 +1142,12 @@ def get_dask_names(ds):
         for k, v in new_dask_names.items():
             assert v == orig_dask_names[k]
 
-        with pytest.raises(ValueError, match=r"some chunks"):
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "chunks keys ('foo',) not found in dataset dimensions ('dim2', 'dim3', 'time', 'dim1')"
```

Comment:
Is it a good idea to test exactly for the error messages? I've done it kind of inconsistently, elsewhere it's not this exact.

Reply:
Generally we don't normally test exactly for the whole error message. Often instead we test for just some key part of the message. So here you might search for "not found in dataset", or "'dim2', 'dim3', 'time', 'dim1'". We don't have any strict rules on this though.

Reply:
I think at least here it's good to test both that 'foo' and the list of dimensions are present in the error message.